src/tmx_json.c

```diff
@@ -205,39 +207,164 @@ static int pjson_tile(json_t *tile_el, tmx_tile **tile_headaddr) {
 	return 1;
 }
 
+
+/* This is a rather slow solution - anyone with more spare time, feel free to rework the whole structure of
+   the tiles list to make this faster */

```

Comment:
A better solution may be to use a hashtable instead of a linked list, feel free to open an issue.